In [9]:
import pandas as pd
 
url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/titanic.csv"
df = pd.read_csv(url)

In [10]:
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [11]:
df.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


**RESPUESTA - Pregunta A**

512.3292 − 31.0000 = 481.3292

Esto muestra que los precios de los boletos eran muy diferentes. La mayoría pagó menos, pero algunos pagaron mucho más, así que las tarifas no eran homogéneas.

**RESPUESTA - Pregunta B**

PassengerId tiene 891 registros, mientras que Age tiene 714. Esto significa que hay 177 pasajeros sin edad registrada, por lo que encontramos datos faltantes.

**RESPUESTA - Pregunta C**

Dos opciones serían:
- Reemplazar las edades faltantes con la mediana o promedio de Age.
- Eliminar los registros que no tengan edad, aunque perderíamos algunos datos.

También se podría estimar la edad usando otros datos, como Pclass o Sex.

**Actividad MIERCOLES 26 de AGOSTO 2026**

Preguntas

1. ¿Cuál fue la tarifa promedio que pagaron las mujeres?
 


2. ¿Y cuál fuel la de los hombres?

In [12]:
tarifa_promedio = df.groupby("Sex")["Fare"].mean()
print(tarifa_promedio)

Sex
female    44.479818
male      25.523893
Name: Fare, dtype: float64


**RSPUESTAS**

R1. female    44.479818

R2. male      25.523893